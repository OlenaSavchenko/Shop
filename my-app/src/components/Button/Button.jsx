import PropTypes from 'prop-types';
import "./Button.scss"

const Button = (props) => {
    const { backgroundColor, text, className, onClick, type } = props
    return (<button
        type={type}
        onClick={onClick}
        style={{ backgroundColor: backgroundColor, color: "white" }}
        className={`btn ${className}`}>{text}
    </button >)
}

Button.propTypes = {
    backgroundColor: PropTypes.string,
    text: PropTypes.node,
    className: PropTypes.string,
    type: PropTypes.string.isRequired,
    onClick: PropTypes.func
}

Button.defaultProps = {
    backgroundColor: "transparent",
    text: null,
    className: "",
    onClick: () => { }

}

export default Button