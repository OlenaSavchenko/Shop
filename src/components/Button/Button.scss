.App__first-btn {
    margin-right: 15px;
  }

.delete-modal__btn {
    padding: 10px 20px;
    &--approve{
        padding: 10px 35px;  
    }
}

.modal-close-icon {
    font-size: 40px;
    line-height: 18px;
    display: block;
    width: 25px;
    height: 25px;
}
