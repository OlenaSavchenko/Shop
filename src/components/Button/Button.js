import PropTypes from 'prop-types';
import "./Button.scss"

const Button = ({ backgroundColor, text, className, onClick, type }) => {
    return (<button
        type={type}
        onClick={onClick}
        style={{ backgroundColor: backgroundColor, color: "white" }}
        className={className ? `btn ${className}` : "btn"}>{text}
    </button >)
}

Button.propTypes = {
    backgroundColor: PropTypes.string,
    text: PropTypes.node,
    className: PropTypes.string,
    type: PropTypes.string.isRequired,
    onClick: PropTypes.func
}

Button.defaultProps = {
    backgroundColor: "transparent",
    text: null,
    className: "",
    onClick: () => { }
}

export default Button