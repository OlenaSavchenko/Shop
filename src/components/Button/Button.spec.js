import { render, screen, fireEvent } from '@testing-library/react';
import Button from "./Button";

describe("Button component", () => {
    it("should render", () => {
        const { getByRole } = render(<Button type="button" />)
        getByRole('button')
    })

    it('should be clicked', () => {
        const handleClick = jest.fn()
        render(<Button type="button" onClick={handleClick} />)
        const el = screen.getByRole('button')
        fireEvent.click(el)
        expect(handleClick).toHaveBeenCalled()
    })

    it('should matches snapshot', () => {
        const { getByRole } = render(<Button type="button" />)
        const el = getByRole('button')
        expect(el).toMatchSnapshot()
    })
})

