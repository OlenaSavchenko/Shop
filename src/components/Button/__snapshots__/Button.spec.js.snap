// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should matches snapshot 1`] = `
<button
  class="btn"
  style="background-color: transparent; color: white;"
  type="button"
/>
`;
